etl/pypasar/db/sql/postgres/drug_exposure/temp_drugmed_view.sql

```diff
@@ -0,0 +1,232 @@
+-- ***********************************************************************
+-- NAME: temp_drug_med_view.sql
+-- Create temp view that joins intraop.drugmed with source_to_concept_map
+-- ***********************************************************************
+
+
+create or replace view {OMOP_SCHEMA}.{DRUGMED_STCM_VIEW} as
+with 
+dm as (
+	select 
+		id,
+		anon_case_no_clindoc,
+		anon_case_no,
+		session_id,
+		operation_id,
+		medication_startdate as "drug_exposure_start_date",
+		medication_startdatetime as "drug_exposure_start_datetime",
+		operation_enddate as "drug_exposure_end_date",
+		(operation_enddate::text || ' ' || operation_endtime)::timestamp as "drug_exposure_end_datetime",
+		dosage_individual as "quantity",
+		medication_name as "medication_name",
+		drug_name as "drug_name",
+		case 
+			when right(upper(medication_name), 2) = '~T' then upper(left(medication_name, -2)) -- 65 records
+			else upper(medication_name)
+		end
+			as "standardized_medication_name",
+		
+		'intraop' as "source_schema",
+		'drugmed' as "source_table"
+	from {INTRAOP_SCHEMA}.drugmed -- 16962 records
+	where medication_name is not null -- 53 records
+    ), -- 16909 records
+
+stcm as (
+	select 
+		row_number() over (partition by source_code order by target_concept_id) as "source_code_rn",
+		regexp_replace(source_code, '\s*\(.*$', '', 'g') as "base_source_code",
+		* 
+	from {OMOP_SCHEMA}.source_to_concept_map stcm
+	where source_vocabulary_id = 'SG_PASAR_INTRAOP_DRUG_MED' 
+    ), -- 83 records
+
+
+-- Remove records with double entry in the join column of dm for sanity check
+dm_non_duplicated as (
+	select 
+		regexp_replace(standardized_medication_name, '\s*\(.*$', '', 'g') as "base_medication_name",
+		* 
+	from dm 
+	where dm.standardized_medication_name not in ('AUGMENTIN (GM)', 'CEFAZOLIN (G)', 'CEPHAZOLIN (G)', 'BUPIVACAINE 0.5% WITH DEXTROSE (MG)', 'PROPOFOL/LIGNOCAINE (MG)', 'TAZOCIN-PIPERACILLIN,TAZOBACTAM (G)')
+    ), -- 16134 records
+
+-- Remove records with double entry in the join column of stcm for sanity check
+stcm_non_duplicated as (
+	select *
+	from stcm
+	where stcm.source_code not in ('AUGMENTIN (GM)', 'CEFAZOLIN (G)', 'CEPHAZOLIN (G)', 'BUPIVACAINE 0.5% WITH DEXTROSE (MG)', 'PROPOFOL/LIGNOCAINE (MG)', 'TAZOCIN-PIPERACILLIN,TAZOBACTAM (G)') --775 records
+	and stcm.source_code_rn = 1 -- ensure unique records
+    ), -- 71 records (83-12) each of these source code is duplicated
```

Comment:
I was doing some checking and looks like the `target_concept_id` is different for the same `source_code`. We will have to clarify this with the Vocab team. Before we proceed to choose the 1st record by default. However this is fine for the 1st draft.
![image](https://github.com/user-attachments/assets/b3d35e41-af20-409e-a492-f26be612d1ed)

Reply:
@alicia-koh This is the response from the vocab team. For each source drug will have two equivalent records in the drug_exposure table.
![image](https://github.com/user-attachments/assets/f74681a6-79b7-461b-8cc0-9fa1a4eec139)

Reply:
NIT: Understand. But it would be great if its possible to handle them generically instead of explicitly checking for specific source codes. Because we expect more codes from vocab team for drug exposure. Then It still involves checking the stcm and handling those specifically. Or do you think there's some other issue in taking this approach.

SQL to check for the multiple target concept ids (would probably just need to filter for drug exposure specific vocabularies)
```
select
   c.source_code,
   c.source_code_description,
   c.source_vocabulary_id,
   c.target_concept_id 
from
   omop_sqldev_schema.source_to_concept_map c 
   inner join
      (
         select
            source_code,
            source_vocabulary_id,
            count(target_concept_id) 
         from
            (
               select
                  source_code,
                  source_vocabulary_id,
                  target_concept_id,
                  source_code_description,
                  ROW_NUMBER() OVER( PARTITION BY source_code, source_vocabulary_id 
               ORDER BY
                  target_concept_id ) rownum 
               from
                  omop_sqldev_schema.source_to_concept_map 
               group by
                  source_code,
                  source_vocabulary_id,
                  target_concept_id,
                  source_code_description 
            )
            b 
         group by
            source_code,
            source_vocabulary_id 
         having
            count(target_concept_id) > 1 
      )
      b 
      on c.source_code = b.source_code 
      and c.source_vocabulary_id = b.source_vocabulary_id 
order by
   source_code
```